apps/teams-test-app/src/components/privateApis/MonetizationAPIs.tsx

```diff
@@ -12,20 +12,25 @@ const CheckMonetizationCapability = (): React.ReactElement =>
   });
 
 const OpenPurchaseExperience = (): React.ReactElement =>
-  ApiWithoutInput({
+  ApiWithTextInput<monetization.PlanInfo | undefined>({
     name: 'monetization_openPurchaseExperience',
     title: 'Open Purchase Experience',
-    onClick: async setResult => {
-      const callback = (error: SdkError | null): void => {
-        if (error) {
-          setResult(JSON.stringify(error));
-        } else {
-          setResult('Success');
+    onClick: {
+      validateInput: planInfo => {
+        if (!planInfo) {
+          return; //This API allow for the input not to be provided
         }
-      };
-
-      monetization.openPurchaseExperience(callback);
-      return 'monetization.openPurchaseExperience()' + noHostSdkMsg;
+        if (typeof planInfo.planId !== 'string') {
```

Comment:
When I was transitioning all the APIs to this new pattern, I decided against checking the type for each of these properties. So I would probably just do:

```
if (!planInfo.planId || !planInfo.term) {
    throw new Error('planId and term are required on input, if provided');
}
```